src/routes/search.js

```diff
@@ -102,32 +117,49 @@ class SearchRoute extends Component {
   }
 
   handleSearch = (search) => {
-    const {
-      match: { params: { searchType } },
-      search: { [searchType]: { query } }
-    } = this.props;
-
+    let searchType = this.getSearchType();
+    if (!searchType) return;
+    const { search: { [searchType]: { query } } } = this.props;
     this.performSearch({ ...query, search });
   };
 
   render() {
-    const { match: { params: { searchType } }, search } = this.props;
-    const { [searchType]: { query, isResolved, isPending, isRejected } } = search;
+    let { location, search, children } = this.props;
+    let searchType = this.getSearchType();
 
-    const searchTypeLocations = Object.keys(search).reduce((locations, type) => ({
+    if (!searchType) return children;
+
+    let { [searchType]: { query, isResolved, isPending, isRejected } } = search;
+    let searchTypeLocations = Object.keys(search).reduce((locations, type) => ({
       ...locations,
       [type]: {
-        pathname: `/eholdings/search/${type}`,
-        search: queryString.stringify(search[type].query)
+        pathname: location.pathname,
+        search: queryString.stringify({
+          ...search[type].query,
+          searchType: type
+        })
       }
     }), {});
 
+    let Results = null;
+    if (searchType === 'vendors') {
+      Results = VendorSearchResultsRoute;
```

Comment:
I don't think that any of these are actual routes any more, right? They're just pure function components. Probably better to put them as just that.

Reply:
I actually already did that in another branch I'm working on for the async abstraction. Moved those changes over to this PR. 👍